File: command/command/document/DocumentTitleMemento.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace command.document
{
	public class DocumentTitleMemento : IMemento
	{
		private readonly IDocument _document;
		private readonly string _oldTitle;

		public DocumentTitleMemento(IDocument document, string oldTitle)
		{
			_document = document ?? throw new ArgumentNullException(nameof(document));
			_oldTitle = oldTitle ?? throw new ArgumentNullException(nameof(oldTitle));
		}

		public void Restore() => _document.Title = _oldTitle;
	}
}

File: command_java/Document/src/document/documentItem/title/DocumentTitleMemento.java
package document.documentItem.title;

import hsitory.IMemento;

public class DocumentTitleMemento implements IMemento {

    public DocumentTitleMemento(String oldTitle, IMutableDocumentTitle documentTitle) {
        _oldTitle = oldTitle;
        _documentTitle = documentTitle;
    }

    @Override
    public void Restore() {
        _documentTitle.SetTitle(_oldTitle);
    }

    private final String _oldTitle;
    private final IMutableDocumentTitle _documentTitle;
}
